components/token/TokenAvatarSet.tsx

```diff
@@ -1,57 +1,96 @@
-import { Avatar, Flex, FlexProps } from '@chakra-ui/react';
-import { AvatarProps } from '@chakra-ui/avatar/src/avatar';
+import {
+    Avatar,
+    Flex,
+    FlexProps,
+    Popover,
+    PopoverTrigger as OrigPopoverTrigger,
+    PopoverContent,
+} from '@chakra-ui/react';
 import { useGetTokens } from '~/lib/global/useToken';
 import Jazzicon, { jsNumberForAddress } from 'react-jazzicon';
+import { PoolTokenPill } from '~/components/token/PoolTokenPill';
+
+export interface TokenData {
+    address: string;
+    weight: string;
+}
 
 interface Props extends FlexProps {
-    addresses: string[];
+    tokenData?: TokenData[];
```

Comment:
don't do it like this, just make everything use the `tokenData` model, and make the weight optional